### ISSUE TITLE
90-alsa-restore.rules causes unwanted volume restore

### ISSUE THREAD

**Issue by hrdl-github** [alsa-project/alsa-utils#244]
Since ef0e588c76fbad4112193d311e51a60d18b44282, `alsactl restore $attr{device/number}` has been used instead of `alsactl restore $attr{number}`. On my system this results in an empty expansion, causing unwanted changes in volume (see https://gitlab.freedesktop.org/pipewire/wireplumber/-/issues/516). The `card*` kernel does have a `number` attribute though:

```
  looking at device '/devices/pci0000:00/0000:00:1f.3/sound/card1/controlC1':
    KERNEL=="controlC1"
    SUBSYSTEM=="sound"
    DRIVER==""
    ATTR{power/control}=="auto"
    ATTR{power/runtime_active_time}=="0"
    ATTR{power/runtime_status}=="unsupported"
    ATTR{power/runtime_suspended_time}=="0"

  looking at parent device '/devices/pci0000:00/0000:00:1f.3/sound/card1':
    KERNELS=="card1"
    SUBSYSTEMS=="sound"
    DRIVERS==""
    ATTRS{id}=="PCH"
    ATTRS{number}=="1"
    ATTRS{power/control}=="auto"
    ATTRS{power/runtime_active_time}=="0"
    ATTRS{power/runtime_status}=="unsupported"
    ATTRS{power/runtime_suspended_time}=="0"
```

**Comment by perexg** [comment]
A proposed fix (we are using this in Fedora for ages):

```
diff --git a/alsactl/90-alsa-restore.rules.in b/alsactl/90-alsa-restore.rules.in
index c0c1b23..eae47b7 100644
--- a/alsactl/90-alsa-restore.rules.in
+++ b/alsactl/90-alsa-restore.rules.in
@@ -2,7 +2,7 @@ ACTION=="add", SUBSYSTEM=="sound", KERNEL=="controlC*", KERNELS!="card*", TEST==
 GOTO="alsa_restore_end"
 
 LABEL="alsa_restore_go"
-TEST!="@daemonswitch@", RUN+="@sbindir@/alsactl restore $attr{device/number}"
-TEST=="@daemonswitch@", RUN+="@sbindir@/alsactl nrestore $attr{device/number}"
+TEST!="@daemonswitch@", RUN+="@sbindir@/alsactl restore /dev/$name"
+TEST=="@daemonswitch@", RUN+="@sbindir@/alsactl nrestore /dev/$name"
 
 LABEL="alsa_restore_end"
```

@tiwai : Comments?
@hrdl-github : Could you test?

**Comment by perexg** [comment]
Yes, $devnode is even better, thanks. Fixed in b5591747fd432d6289c9594a481982a77d357ed5 .
